packages/core/src/execution-engine/workflow-execute.ts

```diff
@@ -1015,8 +1033,14 @@ export class WorkflowExecute {
 
 		const nodeType = workflow.nodeTypes.getByNameAndVersion(node.type, node.typeVersion);
 
+		const routingExecuteFunction = this.getRoutingNodeExecuteFunction(node, nodeType);
+
 		let connectionInputData: INodeExecutionData[] = [];
-		if (nodeType.execute || (!nodeType.poll && !nodeType.trigger && !nodeType.webhook)) {
+		if (
+			nodeType.execute ||
```

Comment:
Would it simplify the implementation if we override `nodeType.execute` with this custom function?

Reply:
we could not do that, it will causes issue for other node's opearations